bool state
---
int32 to_encoder_left
int32 to_encoder_right
int32 to_encoder_accumulate_count
